activities/Chess.activity/js/chessGame.js

```diff
@@ -391,18 +391,24 @@ var ChessGame = {
       var start_time = Date.now();
       var possibleMove = this.state.findmove(this.level);
       var delta = Date.now() - start_time;
+
+      // console.log(`Random Move: ${possibleMove}, Delta: ${delta}`);
+
       // game over
-      if (possibleMove.length === 0) {
+      if (possibleMove.length === 0 && (move.flags === 7 || move.flags === 15 || move.flags === 23 || move.flags === 39)) {
```

Comment:
It seems that instead of testing each value you could check `move.flags & 7`

Reply:
Yeah no, so basically I changed this but the previous issue was due to the test like this , so for example a valid move has a flag 1, and check has 2 , and if it's mate/statemate it's 4, now the flag we get is the OR of all of them. So keeping 
move.flag & 7 will give 1 on a valid move as well which is incorrect giving false positives for draw. + 8 is for capture, and all the cases are covered here, but I did re write the code a bit more after this too.